subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/mvnsettings/DefaultLocalMavenRepositoryLocatorTest.groovy

```diff
@@ -107,29 +107,38 @@ class DefaultLocalMavenRepositoryLocatorTest extends Specification {
 
     def "returns default location if user settings file specifies empty local repository"() {
         when:
-        writeSettingsFile(locations.userSettingsFile, "")
+        writeSettingsFile(locations.userSettingsFile, localRepository)
         1 * system.getProperty("user.home") >> userHome1.absolutePath
 
         then:
         locator.localMavenRepository == defaultM2Repo
+
+        where:
+        localRepository << ["<localRepository/>", "<localRepository></localRepository>", ""]
```

Comment:
I feel like there could be some sort of little helper class that provides these tags to avoid duplication. Not strictly required for getting this PR done though.

```
class LocalRepositoryTagCreator {
    static String getEmptyString() { ... }
    static String getNoValue() { ... }
    static String getForValue(String value) { ... }
}
```